### PR TITLE
add documentation

### PR DESCRIPTION
**Pull request by alberhander**
https://app.shortcut.com/cartoteam/story/240382/define-and-test-high-availability-configuration

### CODE REVIEW

**Comment by alberhander** [comment]
## Tested the config
1. Make sure the service replicas are deployed to different nodes:
```bash
albertoh@albertoh:~/git/carto-selfhosted-helm$ kubectl get pod -o=custom-columns=NODE:.spec.nodeName,NAME:.metadata.name | grep api
...
gke-gke-infra-team-2941-pool-03-2b7e38c3-28lh   k6-perf-tests-carto-maps-api-7c7446775d-c4t4n
gke-gke-infra-team-2941-pool-03-2b7e38c3-5dvf   k6-perf-tests-carto-maps-api-7c7446775d-wlnrq
gke-gke-infra-team-2941-pool-03-2b7e38c3-5dvf   k6-perf-tests-carto-workspace-api-664987f9c5-6hgw5
gke-gke-infra-team-2941-pool-03-2b7e38c3-xwt5   k6-perf-tests-carto-workspace-api-664987f9c5-w7frn
```

2. Make sure the number of replicas scales up according to the rule set, under some stress traffic:
```
albertoh@albertoh:~$ kubectl get hpa
NAME                                REFERENCE                                      TARGETS   MINPODS   MAXPODS   REPLICAS   AGE
k6-perf-tests-carto-maps-api        Deployment/k6-perf-tests-carto-maps-api        6%/50%    2         3         2          174m
k6-perf-tests-carto-workspace-api   Deployment/k6-perf-tests-carto-workspace-api   79%/50%   2         3         3          174m
```

**Comment by alberhander** [comment]
## Updating HA config
After having increased the `maxReplicas` of workspace-api from 3 to 6 for some tests, I encountered an issue: when `maxReplicas > number of nodes` and the application needs to scale up, a hard check `requiredDuringSchedulingIgnoredDuringExecution` won't work, which was my initial suggestion, because the control plane won't allocate more than 1 pod in the same node. The rest of new pods will be stuck in `Pending`:
```yaml
affinity:
    podAntiAffinity:
      requiredDuringSchedulingIgnoredDuringExecution:
        - labelSelector:
            matchExpressions:
              - key: "app.kubernetes.io/component"
                operator: In
                values:
                  - router
          topologyKey: "kubernetes.io/hostname"
```

Instead, using the soft check `preferredDuringSchedulingIgnoredDuringExecution`, the control plane will allocate the "extra pods", even if more than 1 pod ends up in the same node:
```yaml
affinity:
    podAntiAffinity:
      preferredDuringSchedulingIgnoredDuringExecution:
        - weight: 1
          podAffinityTerm:
            labelSelector:
              matchExpressions:
                - key: "app.kubernetes.io/component"
                  operator: In
                  values:
                    - workspace-api
            topologyKey: "kubernetes.io/hostname"
```

Test that it works:
```bash
# service is going to scale up
albertoh@albertoh:~/git/carto-selfhosted-helm$ kubectl get hpa | grep workspace-api
k6-perf-tests-carto-workspace-api   Deployment/k6-perf-tests-carto-workspace-api   182%/75%   2         6         2          27h
$
# service scaled up to 5 replicas
albertoh@albertoh:~/git/carto-selfhosted-helm$ kubectl get hpa | grep workspace-api
k6-perf-tests-carto-workspace-api   Deployment/k6-perf-tests-carto-workspace-api   73%/75%   2         6         5          27h
$
# replicas are allocated covering the 3 nodes
albertoh@albertoh:~/git/carto-selfhosted-helm$ kubectl get pods -o wide --sort-by="{.spec.nodeName}" | grep workspace-api
k6-perf-tests-carto-workspace-api-c695c6c5b-72tln          1/1     Running   0          4m18s   192.168.64.195   gke-gke-infra-team-2941-pool-03-2b7e38c3-5dvf   <none>           <none>
k6-perf-tests-carto-workspace-api-c695c6c5b-tjvpp          1/1     Running   0          3m39s   192.168.64.184   gke-gke-infra-team-2941-pool-03-2b7e38c3-28lh   <none>           <none>
k6-perf-tests-carto-workspace-api-c695c6c5b-twmbm          1/1     Running   0          39s     192.168.64.185   gke-gke-infra-team-2941-pool-03-2b7e38c3-28lh   <none>           <none>
k6-perf-tests-carto-workspace-api-c695c6c5b-f8ndg          1/1     Running   0          39s     192.168.64.2     gke-gke-infra-team-2941-pool-03-2b7e38c3-xwt5   <none>           <none>
k6-perf-tests-carto-workspace-api-c695c6c5b-cgqvd          0/1     Running   0          24s     192.168.64.3     gke-gke-infra-team-2941-pool-03-2b7e38c3-xwt5   <none>           <none>
```

**Comment by alberhander** [comment]
## increasing `maxReplicas` in maps/workspace apis
Please, read [this comment](https://app.shortcut.com/cartoteam/story/237571/revisit-resources-requests-figures-parameters-to-decrease-the-hardware-requirements-when-the-app-is-starting#) for more context.

TLDR: After testing the self-hosted with this configuration + recommended autoscaling HA, I obtained by far the best results with K6.

**Comment by alberhander** [comment]
https://app.shortcut.com/cartoteam/story/237571/revisit-resources-requests-figures-parameters-to-decrease-the-hardware-requirements-when-the-app-is-starting#

**Review by alvarorm22 — APPROVED** [review]
LGTM
